zanata-war/src/main/java/org/zanata/action/MergeTranslationsManager.java

```diff
@@ -0,0 +1,113 @@
+@Slf4j
+public class MergeTranslationsManager implements Serializable {
+    @In
+    private AsyncTaskHandleManager asyncTaskHandleManager;
+
+    @In
+    private MergeTranslationsService mergeTranslationsServiceImpl;
+
+    @In
+    private Identity identity;
+
+    /**
+     * Merge translations from an existing version to another.
+     *
+     * @param sourceProjectSlug - source project identifier
+     * @param sourceVersionSlug - source version identifier
+     * @param targetProjectSlug - target project identifier
+     * @param targetVersionSlug - target version identifier
+     * @param useNewerTranslation - to override translated/approved string
+     *                                 in target with newer entry in source
+     */
+    public void startMergeTranslations(String sourceProjectSlug,
+            String sourceVersionSlug, String targetProjectSlug,
+            String targetVersionSlug, boolean useNewerTranslation) {
+
+        MergeTranslationsKey key =
+                MergeTranslationsKey.getKey(targetProjectSlug,
+                    targetVersionSlug);
+
+        MergeTranslationsTaskHandle handle = new MergeTranslationsTaskHandle();
+        asyncTaskHandleManager.registerTaskHandle(handle, key);
+        mergeTranslationsServiceImpl.startMergeTranslations(sourceProjectSlug,
+                sourceVersionSlug, targetProjectSlug, targetVersionSlug,
+            useNewerTranslation, handle);
+    }
+
+    /**
+     * Cancel running merge translations task
+     *
+     * @param projectSlug - target project identifier
+     * @param versionSlug - target version identifier
+     */
+    public void cancelMergeTranslations(String projectSlug, String versionSlug) {
+        if (isMergeTranslationsRunning(projectSlug, versionSlug)) {
+            MergeTranslationsTaskHandle handle =
+                getMergeTranslationsProcessHandle(projectSlug, versionSlug);
+            handle.cancel(true);
+            handle.setCancelledTime(System.currentTimeMillis());
+            handle.setCancelledBy(identity.getCredentials().getUsername());
+
+            log.info("Merge translations cancelled- {}:{}", projectSlug,
+                    versionSlug);
+        }
+    }
+
+    public MergeTranslationsTaskHandle getMergeTranslationsProcessHandle(
+            String projectSlug, String versionSlug) {
+        return (MergeTranslationsTaskHandle) asyncTaskHandleManager
+                .getHandleByKey(MergeTranslationsKey.getKey(projectSlug, versionSlug));
+    }
+
+    public boolean isMergeTranslationsRunning(String projectSlug, String versionSlug) {
+        MergeTranslationsTaskHandle handle =
+            getMergeTranslationsProcessHandle(projectSlug, versionSlug);
+        return handle != null && !handle.isDone();
+    }
+
+    /**
+     * Key used for copy version task
+     *
+     */
+    @EqualsAndHashCode
+    @Getter
+    @AllArgsConstructor
+    public static final class MergeTranslationsKey implements Serializable {
```

Comment:
All the methods and this class could all have shorter names since they are in `MergeTranslationsManager`. Could just call them `start`, `cancel`, `getProcessHandle`, `isRunning` and `Key`.